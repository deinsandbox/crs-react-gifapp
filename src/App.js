import React, { useState } from "react";
import "./App.css";
import "animate.css";

import { removeSpaces } from "./helpers/strings";

import AddCategory from "./components/AddCategory";
import Grid from "./components/Grid";

function App({ categories = [] }) {
  const [category, setCategory] = useState(categories);

  return (
    <>
      <h2>Gif Expert App</h2>

      <AddCategory setCategory={setCategory} />

      <hr />

      <ol>
        {category.map((item, index) => (
          <Grid key={`${removeSpaces(item)}`} category={item} />
        ))}
      </ol>
    </>
  );
}

export default App;
