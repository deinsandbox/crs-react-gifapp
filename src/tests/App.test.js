import React from "react";
import "@testing-library/jest-dom";

import { shallow } from "enzyme";
import App from "../App";

describe("Test on <App /> component", () => {
  test("should render the component correctly", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should create a list of categories", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<App categories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("Grid").length).toBe(categories.length);
  });
});
