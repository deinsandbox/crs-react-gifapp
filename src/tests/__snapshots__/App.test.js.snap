// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <App /> component should create a list of categories 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <Grid
      category="One Punch"
      key="OnePunch"
    />
    <Grid
      category="Dragon Ball"
      key="DragonBall"
    />
  </ol>
</Fragment>
`;

exports[`Test on <App /> component should render the component correctly 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
